src/kakarot/accounts/eoa/aa/library.cairo

```diff
@@ -165,15 +207,38 @@ namespace ExternallyOwnedAccount {
         pedersen_ptr: HashBuiltin*,
         bitwise_ptr: BitwiseBuiltin*,
         range_check_ptr,
-    }(msg_hash: Uint256, r: Uint256, s: Uint256, v: felt, eth_address: felt) -> (is_valid: felt) {
+    }(
+        msg_hash: Uint256,
+        sub_items: RLP.Item*,
+        eth_address: felt,
+        tx_type: felt,
+        keccak_ptr: felt*,
+        keccak_ptr_start: felt*,
+    ) -> (is_valid: felt) {
         alloc_locals;
-        let (keccak_ptr: felt*) = alloc();
-        local keccak_ptr_start: felt* = keccak_ptr;
-        with keccak_ptr {
-            verify_eth_signature_uint256(msg_hash=msg_hash, r=r, s=s, v=v, eth_address=eth_address);
+        if (tx_type == 2) {
```

Comment:
so `type = 2` introduces only a shift (3) in the indexes `V_IDX`, `R_IDX` and `S_IDX`. We could better just compute the correct indices and avoid the `if/else` and duplication.

```
let shift = tx_type / 2 * 3
let v_idx = V_IDX - shift
...
```